amp_conf/htdocs/admin/libraries/BMO/Freepbx_conf.class.php

```diff
@@ -377,10 +377,42 @@ function parse_amportal_conf($filename, $bootstrap_conf = array(), $file_is_auth
 		  switch ($arr[0]) {
 			  // for type dir, make sure there is no trailing '/' to keep consistent everwhere
 			  //
-        case CONF_TYPE_DIR:
-				  if (!isset($this->conf[$key]) || trim($this->conf[$key]) == '') {
-					  $this->conf[$key] = $arr[1];
-				  } else {
+		    case CONF_TYPE_DIR:
+			  if (!isset($this->conf[$key]) || trim($this->conf[$key]) == '') {
+				$this->conf[$key] = $arr[1];
+				if (PHP_OS == 'FreeBSD') {
```

Comment:
Ok so we can't have this here. This should be set either in install or the installer files. Setting hard coded paths in here is bad because all values are retrieved from the database, if anyone goes to change anything you've now hard set all values.